CMakeLists.txt

```diff
@@ -1,36 +1,17 @@
 cmake_minimum_required(VERSION 3.4.3)
 
-project(OptSched)
 
 option(OPTSCHED_INCLUDE_TESTS "Generate build targets for the OptSched unit tests." ON)
-option(OPTSCHED_ENABLE_AMDGPU "Build the AMDGPU code. Requires that the AMDGPU target is supported." OFF)
+set(OPTSCHED_ENABLE_AMDGPU "Build the AMDGPU code. Requires that the AMDGPU target is supported." ON)
 set(OPTSCHED_LIT_ARGS "-sv" CACHE STRING "Arguments to pass to lit")
 set(OPTSCHED_EXTRA_LINK_LIBRARIES "" CACHE STRING "Extra link_libraries to pass to OptSched, ;-separated")
 set(OPTSCHED_EXTRA_INCLUDE_DIRS "" CACHE STRING "Extra include_directories to pass to OptSched, ;-separated")
 # To add OptSched debug defines, e.g.:
 # '-DOPTSCHED_EXTRA_DEFINITIONS=-DIS_DEBUG_DEFS_AND_USES;-DIS_DEBUG_DEF_USE_COUNT'
 set(OPTSCHED_EXTRA_DEFINITIONS "" CACHE STRING "Extra add_definitions to pass to OptSched, ;-separated")
 
-if(TARGET LLVMCodeGen)
-  set(llvm_subproject TRUE)
-else()
-  set(llvm_subproject FALSE)
-endif()
-
-if(NOT llvm_subproject)
-  set(llvm_version 6.0)
-  if(OPTSCHED_ENABLE_AMDGPU)
-    set(llvm_version 9.0)
-  endif()
+set(llvm_subproject TRUE)
 
-  set(OPTSCHED_LLVM_VERSION ${llvm_version} CACHE STRING "The LLVM version to build OptSched with (independent build only)")
-
-  find_package(LLVM ${OPTSCHED_LLVM_VERSION} REQUIRED CONFIG)
-endif()
-
-if(NOT CMAKE_BUILD_TYPE)
-  message(FATAL_ERROR "Must set cmake build type.")
-endif()
```

Comment:
Don't delete this

Reply:
Modifications in CMakeLists were mainly due to convenience. It's going to need some redesign going forward -- for example, the build target will be LLVMAMDGPUCodeGen.